gerritlab/global_vars.py

```diff
@@ -175,17 +148,11 @@ def _get_private_token(
     raise SystemExit(f"Unable to find private token for {host}")
 
 
-def _get_upstream_branch(repo: Repo) -> "str|None":
-    local_branch_name = repo.head.reference.name
-    with repo.config_reader() as conf:
-        section = f'branch "{local_branch_name}"'
-        try:
-            remote_ref = conf.get(section, "merge")
-        except (configparser.NoSectionError, configparser.NoOptionError):
-            return None
-        m = re.match(r"refs/heads/(.*)$", remote_ref)
-        if not m:
-            raise Exception(
-                f"Unexpected remote tracking branch format: {remote_ref}"
-            )
-        return m[1]
+def _get_default_branch(repo: Repo) -> "str|None":
```

Comment:
To avoid a network round trip here, we could check the `git symbolic-ref refs/remotes/<remote>/HEAD` and fall back to the fetching from remote method. Git should set that on clone.

Reply:
Nice find! If this works in most cases, I wonder if we even need the fallback. If there are edge cases where this doesn't work, we could error out, or fall back to the default set at the top of global_vars.py. In any case, you always have the option of adding a `.gitreview` file.

Reply:
One case I think that this would fail in is when you created the repo on your laptop and pushed it, rather than cloned it to create your local repo (at least that is the case for a repo that I made yesterday, maybe I pushed with weird flags, but I'm pretty sure I copy-pasted what GitLab told me to do to push).

Reply:
Yes I verified that's the case if you create the remote with `git remote add` without the `-t` option ([docs](https://git-scm.com/docs/git-remote)).

I added instructions on how to fix this in the error message that is displayed if the symbolic-ref is not found.